File: M/1033-moving-stones-until-consecutive/main.go
package main

import (
	"fmt"
	"sort"
)

func numMovesStones(a int, b int, c int) []int {
	buf := []int{a, b, c}
	sort.Ints(buf)
	max := buf[2] - buf[1] - 1 + buf[1] - buf[0] - 1
	x, y, z := buf[0], buf[1], buf[2]
	min := 0

	left := y - x
	right := z - y
	if left == 1 && right == 1 {
	} else if left == 1 || right == 1 || right == 2 || left == 2 {
		min = 1
	} else {
		min = 2
	}

	return []int{min, max}
}

func main() {
	fmt.Println("a")
}

File: M/1033-moving-stones-until-consecutive/main.rs
struct Solution {}

impl Solution {
  pub fn num_moves_stones(a: i32, b: i32, c: i32) -> Vec<i32> {
    let mut s: Vec<i32> = vec![a, b, c];
    s.sort();

    vec![
      match (s[1] - s[0], s[2] - s[1]) {
        (1, 1) => 0,
        (1, _) => 1,
        (_, 1) => 1,
        (2, _) => 1,
        (_, 2) => 1,
        (_, _) => 2,
      },
      s[2] - s[0] - 2,
    ]
  }
}
